source/Engine.cpp

```diff
@@ -1444,152 +1453,165 @@ void Engine::CalculateStep()
 
 	// Handle the mouse input of the mouse navigation
 	HandleMouseInput(activeCommands);
-	// Now, all the ships must decide what they are doing next.
-	ai.Step(activeCommands);
 
-	// Clear the active players commands, they are all processed at this point.
-	activeCommands.Clear();
-
-	// Perform actions for all the game objects. In general this is ordered from
-	// bottom to top of the draw stack, but in some cases one object type must
-	// "act" before another does.
-
-	// The only action stellar objects perform is to launch defense fleets.
+	const Ship *flagship = player.Flagship();
 	const System *playerSystem = player.GetSystem();
-	for(const StellarObject &object : playerSystem->Objects())
-		if(object.HasValidPlanet())
-			object.GetPlanet()->DeployDefense(newShips);
 
-	// Keep track of the flagship to see if it jumps or enters a wormhole this turn.
-	const Ship *flagship = player.Flagship();
-	bool flagshipWasUntargetable = (flagship && !flagship->IsTargetable());
-	bool wasHyperspacing = (flagship && flagship->IsEnteringHyperspace());
-	// First, move the player's flagship.
-	if(flagship)
-		MoveShip(player.FlagshipPtr());
-	const System *flagshipSystem = (flagship ? flagship->GetSystem() : nullptr);
-	bool flagshipIsTargetable = (flagship && flagship->IsTargetable());
-	bool flagshipBecameTargetable = flagshipWasUntargetable && flagshipIsTargetable;
-	// Then, move the other ships.
-	for(const shared_ptr<Ship> &it : ships)
+	if (timePaused)
```

Comment:
lots of these, I will not add more

Reply:
Apologies. Will go through them, force of habit from writing on other projects.